platforms/core-execution/workers/src/integTest/groovy/org/gradle/workers/internal/WorkerExecutorParallelBuildOperationsIntegrationTest.groovy

```diff
@@ -163,12 +163,15 @@ This means workTask and slowTask would be expected to run concurrently in this c
         """
 
         then:
-        workTaskDoesNotCompleteFirst()
+        workTaskDoesNotCompleteFirst(true)
     }
 
-    private void workTaskDoesNotCompleteFirst() {
+    private void workTaskDoesNotCompleteFirst(boolean expectDeprecation = false) {
         blockingHttpServer.expectConcurrent("workTask", "slowTask")
         args("--max-workers=4")
+        if (expectDeprecation) {
+            executer.expectDocumentedDeprecationWarning("Listener registration using Gradle.addListener() has been deprecated. This will fail with an error in Gradle 9.0. Consult the upgrading guide for further information: https://docs.gradle.org/current/userguide/upgrading_version_7.html#task_execution_events")
```

Comment:
💅 It'd be clearer to extract this as method and call it from the places that pass `true` to the argument

Reply:
See: https://github.com/gradle/gradle/pull/30000/commits/93031caaa187f54a50768babfe37dcdd293e6faf